packages/pluginutils/types/index.d.ts

```diff
@@ -66,7 +66,7 @@ export function dataToEsm(data: unknown, options?: DataToEsmOptions): string;
  * Extracts the names of all assignment targets based upon specified patterns.
  * @param param An `acorn` AST Node.
  */
-export function extractAssignedNames(param: BaseNode): string[];
```

Comment:
changes between `BaseNode` to `Node` seems to have a but of history here
I saw a few issues and PRs addressing them, seemed like a back and forth

according to recent https://github.com/Rich-Harris/estree-walker/issues/28, using `Node` is the way, at least from the walker perspective

hope it's OK also for `pluginutils`